odoo/models.py

```diff
@@ -3144,6 +3145,14 @@ def unlink(self):
                 if attachments:
                     attachments.unlink()
 
+                # For the same reason, remove the relevant records in ir_translation
```

Comment:
We could add an ``if`` statement that checks whether there is at least one translated field on the model:
```python
if any(field.translate for field in self._fields.values()):
    ...
```
Roughly less than 20% of models have translated fields. This would save a query on 80% of models.